include/sys/util_macro.h

```diff
@@ -0,0 +1,621 @@
+/*
+ * Copyright (c) 2011-2014, Wind River Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Macro utilities
+ *
+ * Macro utilities are the public interface for C/C++ code and device tree
+ * related implementation.  In general, C/C++ will include <sys/util.h> instead
+ * this file directly.  For device tree implementation, this file should be
+ * include instead <sys/util_internal.h>
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_UTIL_MACROS_H_
+#define ZEPHYR_INCLUDE_SYS_UTIL_MACROS_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
```

Comment:
Shouldn't this be taken out to `util.h` to surround the inclusion of `util_macro.h`?
I do not think that the DTS intermediate file will have `extern "C" {` inserted, because it will not have the `__cplusplus` defined, but if this is supposed to have only DTS interpreter acceptable syntax then this block should be thrown outside.
What do you think @galak?

Reply:
I think its ok to leave here since the only time __cplusplus should get defined is with a C++ compiler.  So with DTS this will just go away.